src/components/Vault/VaultWithdrawForm.tsx

```diff
@@ -24,9 +25,13 @@ const VaultWithdrawForm: FC<VaultProps> = (props) => {
   const chainId = useActiveChainId()
   const { address: userAddress } = useAccount()
   const vault = useVault(props)
+  const [invalidateHoldingsVaults, setInvalidateHoldingsVaults] =
+    useState(false)
 
   const underlyingAssets = vault.data?.underlyingAssets
 
+  useInvalidateHoldingsVaults({ enabled: invalidateHoldingsVaults })
```

Comment:
This is kinda complicated. I think it would be cleaner to just invoke the invalidation directly in the submit handler instead of doing this thing with the state variable to trigger this hook.

If you want to abstract the some of the invalidation code away into a hook, I find this pattern of returning a function from a hook to be really powerful:

```ts
export function useInvalidateHoldingsVaults() {
  const queryClient = useQueryClient()
  const invalidateHoldingsVaults = () => {
    queryClient.invalidateQueries(...)
  }
  return invalidateHoldingsVaults
}

// then to use it
const invalidateHoldingsVaults = useInvalidateHoldingsVaults()
invalidateHoldingsVaults()
```

Reply:
that's cool, didn't know it too